clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -254,8 +265,10 @@ private static boolean matchingTopic(FetchResponseData.FetchableTopicResponse pr
     private static FetchResponseData toMessage(Errors error,
                                                int throttleTimeMs,
                                                int sessionId,
-                                               Iterator<Map.Entry<TopicIdPartition, FetchResponseData.PartitionData>> partIterator) {
+                                               Iterator<Map.Entry<TopicIdPartition, FetchResponseData.PartitionData>> partIterator,
+                                               List<Node> nodeEndpoints) {
         List<FetchResponseData.FetchableTopicResponse> topicResponseList = new ArrayList<>();
+        FetchResponseData data = new FetchResponseData();
```

Comment:
nit: can we move the object creation closer to where it's updated and returned at the bottom of the method?

Reply:
sgtm